src/main/Tests/BattleshipModelTest.java

```diff
@@ -63,11 +62,9 @@ void aifire(){
 
     @Test
     void aiFireHard(){
-        Board playerTest = new Board();
-        aiFireHard(playerTest);
-
-        assertNotNull(player.getAllShots().get(0));
-
+        BattleshipModel model = new BattleshipModel();
```

Comment:
The changes to aiFireHard don't actually work, but we'll fix that. The for loop needs to be changed to not go out of bounds.